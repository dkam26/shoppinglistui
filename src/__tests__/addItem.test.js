import React from 'react';
import { shallow, mount } from 'enzyme';
import {configure} from 'enzyme';
import AddItem  from '../components/add.item.component';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import toJson,{ shallowToJson } from 'enzyme-to-json';
import LocalStorageMock from '../setupTests';
import ReactDOM from 'react-dom';
configure({ adapter: new Adapter() });
describe('add item component', () => {
    const props={
        getLists:()=>{},
        componentDidMount:()=>{}
    }
    const wrapper =  shallow(<AddItem />);
    it('should run', () => {
        expect(wrapper.exists(<div></div>)).toBe(true);
    });
    it('should render correctly', () => {
        
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        });
    it('should confirm states',()=>{
        expect(wrapper.state().shoppinglists).toBe(false)
        expect(wrapper.state().item).toBe('')
        expect(wrapper.state().Quantity).toBe('')
        expect(wrapper.state().Amountspent).toBe('')
        expect(wrapper.state().shoplist).toBe('')

    })
    it('render the three inputs', () =>{
            expect(wrapper.find("#item")).toHaveLength(1);
            expect(wrapper.find("#Quantity")).toHaveLength(1);
            expect(wrapper.find("#Amountspent")).toHaveLength(1);
            expect(shallowToJson(wrapper)).toMatchSnapshot();
        }); 
    it('should add item',()=>{
            wrapper.find('#add').simulate('click')
            expect(toJson(wrapper)).toMatchSnapshot();
           
            
        });
    it('should ve the following methods',()=>{
            wrapper.instance().getLists();
            wrapper.instance().componentDidMount();
           
           });
  
})