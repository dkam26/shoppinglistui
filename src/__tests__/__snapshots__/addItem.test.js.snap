// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add item component render the three inputs 1`] = `
<div>
  <Redirect
    push={false}
    to="/"
  />
</div>
`;

exports[`add item component should render correctly 1`] = `
<div>
  <Redirect
    push={false}
    to="/"
  />
</div>
`;
